scene/gui/foldable_container.cpp

```diff
@@ -0,0 +1,659 @@
+
+			Ref<Texture2D> icon = _get_title_icon();
+
+			title_width -= icon->get_width() + h_separation;
+			Point2 icon_pos;
+			icon_pos.x = title_style->get_margin(SIDE_LEFT);
+			icon_pos.y = MAX((title_panel_height - title_ms.height - icon->get_height()) * 0.5, 0) + title_style_ofs;
+
+			bool rtl = is_layout_rtl();
+			if (rtl) {
+				icon_pos.x = size.width - icon_pos.x - icon->get_width();
+			} else {
+				title_pos.x = title_style->get_margin(SIDE_LEFT) + icon->get_width() + h_separation;
+			}
+			icon->draw(ci, title_ofs + icon_pos);
+
+			int offset = 0;
+			title_width -= offset;
+			if (rtl) {
+				title_pos.x = offset + h_separation + title_controls_width;
+			}
+
+			Color font_color = folded ? theme_cache.title_collapsed_font_color : theme_cache.title_font_color;
+			if (is_hovering) {
+				font_color = theme_cache.title_hovered_font_color;
+			}
+			text_buf->set_width(title_width - title_controls_width);
+
+			if (title_width > 0) {
+				if (theme_cache.title_font_outline_size > 0 && theme_cache.title_font_outline_color.a > 0) {
+					text_buf->draw_outline(ci, title_ofs + title_pos, theme_cache.title_font_outline_size, theme_cache.title_font_outline_color);
+				}
+				text_buf->draw(ci, title_ofs + title_pos, font_color);
+			}
+
+			if (!folded) {
+				Rect2 panel_rect = Rect2(Point2(0, (title_position == POSITION_TOP) ? title_panel_height : 0), Size2(size.width, size.height - title_panel_height));
+				if (title_position == POSITION_BOTTOM) {
+					draw_set_transform(Point2(0.0, title_style->get_draw_rect(panel_rect).size.height), 0.0, Size2(1.0, -1.0));
+				}
+				theme_cache.panel_style->draw(ci, panel_rect);
+				if (title_position == POSITION_BOTTOM) {
+					draw_set_transform(Point2(), 0.0, Size2(1.0, 1.0));
+				}
+			}
+
+			if (has_focus()) {
+				Rect2 focus_rect = folded ? Rect2(-title_ofs, Size2(size.width, title_panel_height)) : Rect2(Point2(), size);
+				if (title_position == POSITION_BOTTOM) {
+					draw_set_transform(Point2(0.0, title_style->get_draw_rect(focus_rect).size.height), 0.0, Size2(1.0, -1.0));
+				}
+				theme_cache.focus_style->draw(ci, focus_rect);
+				if (title_position == POSITION_BOTTOM) {
+					draw_set_transform(Point2(), 0.0, Size2(1.0, 1.0));
+				}
+			}
+		} break;
+
+		case NOTIFICATION_SORT_CHILDREN: {
+			bool rtl = is_layout_rtl();
+			const Vector2 size = get_size();
+			const Vector2 title_size = _get_title_panel_min_size();
+			const Ref<StyleBox> title_style = _get_title_style();
+
+			uint32_t title_count = title_controls.size();
+			if (title_count > 0) {
+				real_t offset = 0.0;
+				if (rtl) {
+					offset = title_style->get_margin(SIDE_LEFT);
+				} else {
+					offset = _get_title_controls_width();
+					offset = size.x - title_style->get_margin(SIDE_RIGHT) - offset;
+				}
+
+				real_t v_center = title_size.y * 0.5;
+				for (int i = 0; i < title_count; i++) {
```

Comment:
```suggestion
				for (uint32_t i = 0; i < title_count; i++) {
```
Since `title_count` is `uint32_t` or it will cause "comparison of integers of different signs" error.